include/rmm/device_uvector.hpp

```diff
@@ -71,21 +71,27 @@ class device_uvector {
                 "device_uvector only supports types that are trivially copyable.");
 
  public:
-  using value_type      = T;
-  using size_type       = std::size_t;
-  using reference       = value_type&;
-  using const_reference = value_type const&;
-  using pointer         = value_type*;
-  using const_pointer   = value_type const*;
-  using iterator        = pointer;
-  using const_iterator  = const_pointer;
+  using value_type = T;            ///< Stored value type
+  using size_type  = std::size_t;  ///< The type used for the size of the vector
```

Comment:
Probably for a separate cleanup, but just to note that this class uses `std::size_t` explicitly everywhere instead of (probably) using `size_type`.

Reply:
Please file an issue.

Reply:
#1318.